packages/fxa-auth-server/scripts/remove-accounts.ts

```diff
@@ -0,0 +1,175 @@
+    }
+  }
+
+  return accountsToDelete;
+}
+
+export async function cancelSubscriptionsAndDeleteCustomer(
+  stripeHelper: StripeHelper,
+  paypalHelper: PayPalHelper,
+  account: Account
+): Promise<void> {
+  const stripeCustomer = await stripeHelper.fetchCustomer(account.uid, [
+    'subscriptions',
+  ]);
+
+  if (stripeCustomer) {
+    const paymentProvider = (
+      await stripeHelper.getPaymentProvider(stripeCustomer)
+    ).toLowerCase();
+
+    // Cancel any subscriptions and issue refunds
+    if (stripeCustomer.subscriptions) {
+      for (const subscription of stripeCustomer.subscriptions.data) {
+        await issueRefund(
+          stripeHelper,
+          paypalHelper,
+          subscription,
+          paymentProvider
+        );
+        await stripeHelper.cancelSubscription(subscription.id);
+      }
+    }
+
+    // Remove the customer from stripe
+    await stripeHelper.removeCustomer(account.uid, stripeCustomer.email!);
+
+    // Delete any PayPal billing agreements
+    if (paymentProvider === 'paypal') {
+      await deletePayPalBillingAgreements(paypalHelper, account);
+    }
+  }
+}
+
+export async function issueRefund(
+  stripeHelper: StripeHelper,
+  paypalHelper: PayPalHelper,
+  subscription: Stripe.Subscription,
+  paymentProvider: string
+) {
+  const latestInvoice = await stripeHelper.getInvoice(
+    subscription.latest_invoice as string
+  );
+
+  switch (paymentProvider) {
+    case 'stripe':
+      await stripeHelper.refundPayment(
+        latestInvoice.payment_intent as string,
+        'fraudulent'
+      );
+      break;
+    case 'paypal':
+      const paypalTransactionId =
+        await stripeHelper.getInvoicePaypalTransactionId(latestInvoice);
+      if (paypalTransactionId) {
+        await paypalHelper.issueRefund(latestInvoice, paypalTransactionId);
+      }
+      break;
+    default:
+      break;
+  }
+}
+
+export async function deletePayPalBillingAgreements(
+  paypalHelper: PayPalHelper,
+  account: Account
+) {
+  const paypalAgreements = await getAllPayPalBAByUid(account.uid);
+  const activeIds = paypalAgreements.filter((ba: any) => {
+    ['active', 'pending', 'suspended'].includes(ba.status.toLowerCase());
+  });
+
+  await Promise.allSettled(
+    activeIds.map((ba) => {
+      paypalHelper.cancelBillingAgreement(ba.billingAgreementId);
+    })
+  );
+
+  await deleteAllPayPalBAs(account.uid);
+}
+
+export async function init() {
+  const { log, database, senders, stripeHelper } =
+    await setupProcessingTaskObjects('remove-accounts');
+
+  const paypalClient = new PayPalClient(
+    config.subscriptions.paypalNvpSigCredentials
+  );
+  Container.set(PayPalClient, paypalClient);
+  const paypalHelper = new PayPalHelper({ log });
+  Container.set(PayPalHelper, paypalHelper);
+  const statsd = Container.get(StatsD);
+
+  statsd.increment('remove-accounts.startup');
+
+  // retreive all unverified accounts
+  const allAccounts = await retreiveUnverifiedAccounts(database);
```

Comment:
nit: should this be named `allAccounts` or `allUnverifiedAccounts`?

Reply:
Oh, just to tac onto this, would it make sense to rename the file+script `remove-unverified-accounts` or something similar? Just so its clear what the script does.

Reply:
both updated